addons/sale/models/sale_order_line.py

```diff
@@ -1163,3 +1163,28 @@ def _convert_to_sol_currency(self, amount, currency):
                 round=False,
             )
         return amount
+
+    def action_add_from_catalog(self):
+        order = self.env['sale.order'].browse(self.env.context.get('order_id'))
+        kanban_view_id = self.env.ref('sale.sale_product_catalog_kanban_view').id
+        search_view_id = self.env.ref('sale.sale_product_catalog_search_view').id
+
+        return {
+            'type': 'ir.actions.act_window',
+            'name': _('Products'),
+            'res_model': 'product.product',
+            'views': [(kanban_view_id, 'kanban'), (False, 'form')],
+            'search_view_id': [search_view_id, 'search'],
+            'domain': order._get_product_catalog_domain(),
+            'context': {
+                **self.env.context,
+                'order_id': order.id,
+                'currency_id': order.currency_id.id,
```

Comment:
```suggestion
                'product_catalog_currency_id': order.currency_id.id,
```
to make sure there is no side effect from this context key